packages/core/src/utils/ethapi.ts

```diff
@@ -128,7 +128,8 @@ export class EthApi {
 
     const signerAccount = account || requireAccount(this);
 
-    const response = await this.rpc("eth_sign", [signerAccount, messageHex]);
+    // const response = await this.rpc("eth_sign", [signerAccount, messageHex]);
+    const response = await this.rpc("personal_sign", messageHex, signerAccount);
```

Comment:
Two things here: 

1) The commented-out line was causing an error in MetaMask/Web3's `eth_sign` stack. That code was expecting 2 separate string args, but the commented out line is passing 1 array arg (that has 2 string members). So the first change I made was to change the args that are being passed. Cool?

2) @tobek mentioned that `web3.personal.sign` is better to use for signing, so I made this change. Is that cool / compatible? Or are there any things upstream/downstream that this will break?

@ritave, I think you're also the knowledgeable party here...

Reply:
Right now signing is bit of a mess.
`personal_sign` only works for utf8 strings, which is not the case when for example we're approving articles on newsrooms.

I think for now the library should support both as two API endpoints and use one that works for your use-case

See:
https://github.com/0xProject/0x-monorepo/issues/162
https://github.com/ethereum/EIPs/pull/683
https://medium.com/metamask/the-new-secure-way-to-sign-data-in-your-browser-6af9dd2a1527